contentcuration/contentcuration/urls.py

```diff
@@ -88,14 +101,145 @@ def get_queryset(self):
         return Channel.objects.filter(Q(editors=self.request.user) | Q(viewers=self.request.user) | Q(public=True)).distinct()
 
 
+class ChannelFilter(FilterSet):
+    edit = BooleanFilter(method="filter_edit")
+    view = BooleanFilter(method="filter_view")
+    bookmark = BooleanFilter(method="filter_bookmark")
+
+    class Meta:
+        model = Channel
+        fields = ("edit", "view", "public", "bookmark")
+
+    def filter_edit(self, queryset, name, value):
+        return queryset.filter(editors=self.request.user)
+
+    def filter_view(self, queryset, name, value):
+        return queryset.filter(viewers=self.request.user)
+
+    def filter_bookmark(self, queryset, name, value):
+        return queryset.filter(bookmarked_by=self.request.user)
+
+
+class SQCount(Subquery):
+    # Include ALIAS at the end to support Postgres
+    template = "(SELECT COUNT(%(field)s) FROM (%(subquery)s) AS %(field)s__sum)"
+    output_field = IntegerField()
+
+
+class ChannelSlimViewSet(viewsets.GenericViewSet):
```

Comment:
This viewset is specifically designed to return a list of channels in a single query, and in as performant way as possible, for a single user.

One important thing to note here is that the admin will not see arbitrary channels here, which I think is in line with how it is currently.

All data is fetched in a single (but slightly large query) - by judicious and filtered prefetching, it is hoped that we will avoid combinatorial issues (like channels with large numbers of editors being slower to load because of prefetching all the editors, rather than just the current user who is the main relevant person here).